usr/lib/tpm_stdll/tpm_specific.c

```diff
@@ -2572,6 +2573,53 @@ CK_RV token_specific_des_ecb(STDLL_TokData_t * tokdata,
     }
 
     return rc;
+#else
+    const EVP_CIPHER *cipher = EVP_des_ecb();
+    EVP_CIPHER_CTX *ctx = NULL;
+    CK_ATTRIBUTE *attr = NULL;
+    unsigned char dkey[8];
+    CK_ULONG rc;
+    int outlen;
+
+    UNUSED(tokdata);
+
+    // get the key value
+    if (template_attribute_find(key->template, CKA_VALUE, &attr) == FALSE) {
+        TRACE_ERROR("template_attribute_find(CKA_VALUE) failed.\n");
+        return CKR_FUNCTION_FAILED;
+    }
+
+    memcpy(dkey, attr->pValue, sizeof(dkey));
```

Comment:
RSA testcases on TPM token segfaults here (line 2592):

```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x000003ff94564412 in token_specific_des_ecb (tokdata=<optimized out>,
    in_data=0x3ffe3cfdff0 "\210\065^\202v\250f\371\314\301P\005\201\342\203\302\334\220p\352;\033\003INW\241D\326\370\272\352",
    in_data_len=32, out_data=0x3ffe3cfe010 "", out_data_len=out_data_len@entry=0x3ffe3cfdf50, key=0x293d9d10, encrypt=0 '\000')
    at usr/lib/tpm_stdll/tpm_specific.c:2592
2592        memcpy(dkey, attr->pValue, sizeof(dkey));
Missing separate debuginfos, use: dnf debuginfo-install glibc-2.27-32.1.ibm.fc28.s390x
(gdb) bt
#0  0x000003ff94564412 in token_specific_des_ecb (tokdata=<optimized out>,
    in_data=0x3ffe3cfdff0 "\210\065^\202v\250f\371\314\301P\005\201\342\203\302\334\220p\352;\033\003INW\241D\326\370\272\352",
    in_data_len=32, out_data=0x3ffe3cfe010 "", out_data_len=out_data_len@entry=0x3ffe3cfdf50, key=0x293d9d10, encrypt=0 '\000')
    at usr/lib/tpm_stdll/tpm_specific.c:2592
#1  0x000003ff94553f5c in ckm_des_ecb_decrypt (tokdata=<optimized out>, in_data=<optimized out>, in_data_len=<optimized out>,
    out_data=<optimized out>, out_data_len=0x3ffe3cfdf50, key=0x293d9d10) at usr/lib/common/mech_des.c:1464
#2  0x000003ff945540e6 in des_ecb_decrypt (tokdata=0x28ff3c20, sess=<optimized out>, length_only=<optimized out>,
    ctx=<optimized out>,
    in_data=0x3ffe3cfdff0 "\210\065^\202v\250f\371\314\301P\005\201\342\203\302\334\220p\352;\033\003INW\241D\326\370\272\352",
    in_data_len=32, out_data=0x3ffe3cfe010 "", out_data_len=0x3ffe3cfdf50) at usr/lib/common/mech_des.c:127
#3  0x000003ff9454b34a in decr_mgr_decrypt (tokdata=0x28ff3c20, sess=0x290ed0f0, length_only=<optimized out>, ctx=0x290ed170,
    in_data=0x3ffe3cfdff0 "\210\065^\202v\250f\371\314\301P\005\201\342\203\302\334\220p\352;\033\003INW\241D\326\370\272\352",
    in_data_len=32, out_data=0x3ffe3cfe010 "", out_data_len=0x3ffe3cfdf50) at usr/lib/common/decr_mgr.c:694
#4  0x000003ff9451e65e in SC_Decrypt (tokdata=0x28ff3c20, sSession=sSession@entry=0x3ffe3cfde20,
    pEncryptedData=pEncryptedData@entry=0x3ffe3cfdff0 "\210\065^\202v\250f\371\314\301P\005\201\342\203\302\334\220p\352;\033\003INW\241D\326\370\272\352", ulEncryptedDataLen=ulEncryptedDataLen@entry=32, pData=pData@entry=0x3ffe3cfe010 "", pulDataLen=0x3ffe3cfdf50)
    at usr/lib/common/new_host.c:2324
#5  0x000003ff9528433e in C_Decrypt (hSession=<optimized out>,
    pEncryptedData=0x3ffe3cfdff0 "\210\065^\202v\250f\371\314\301P\005\201\342\203\302\334\220p\352;\033\003INW\241D\326\370\272\352", ulEncryptedDataLen=32, pData=0x3ffe3cfe010 "", pulDataLen=0x3ffe3cfdf50) at usr/lib/api/api_interface.c:395
#6  0x0000000001009d2c in do_WrapUnwrapRSA (tsuite=tsuite@entry=0x10384e0 <generated_keywrap_test_suites>)
    at testcases/crypto/rsa_func.c:1315
#7  0x000000000100c7f8 in rsa_funcs () at testcases/crypto/rsa_func.c:1752
#8  0x0000000001000dfe in main (argc=<optimized out>, argv=<optimized out>) at testcases/crypto/rsa_func.c:1805
```
attr->pValue is NULL:
`*attr = {type = 17, pValue = 0x0, ulValueLen = 0}`

This is the same error as mentioned here: https://github.com/opencryptoki/opencryptoki/issues/280#issuecomment-600105191
So although its in your modified code, ist still he same old error, not cuased by your changes.

However, AES, DES, 3DES tests run fine.